optaconf-webapp/src/main/java/org/optaconf/service/ScheduleService.java

```diff
@@ -1,113 +1,134 @@
-    }
-
-    @GET // TODO should be post
-    @Path("/solve")
-    @Produces("application/json")
-    public String solveSchedule(@PathParam("conferenceId") Long conferenceId) {
-        Solver oldSolver = scheduleManager.getSolver();
-        if (oldSolver != null && oldSolver.isSolving()) {
-            oldSolver.terminateEarly();
-        }
-        Solver solver = solverFactory.buildSolver();
-        // TODO Use async solving https://developer.jboss.org/message/910391
-//        scheduleManager.setSolver(solver);
-//        executor.submit(new SolverCallable(solver, scheduleManager.getSchedule()));
-//        return "Solved started.";
-        solver.solve(scheduleManager.getSchedule());
-        scheduleManager.setSchedule((Schedule) solver.getBestSolution());
-        return "Solved. Sorry it took so long";
-    }
-
-    @GET
-    @Path("/isSolving")
-    @Produces("application/json")
-    public boolean isSolving(@PathParam("conferenceId") Long conferenceId) {
-        Solver solver = scheduleManager.getSolver();
-        return solver != null && solver.isSolving();
-    }
-
-    @GET // TODO should be post
-    @Path("/terminateSolving")
-    @Produces("application/json")
-    public void terminateSolving(@PathParam("conferenceId") Long conferenceId) {
-        Solver solver = scheduleManager.getSolver();
-        if (solver != null) {
-            solver.terminateEarly();
-        }
-    }
-
-    private class SolverCallable implements Runnable {
-
-        private final Solver solver;
-        private final Schedule schedule;
-
-        private SolverCallable(Solver solver, Schedule schedule) {
-            this.solver = solver;
-            this.schedule = schedule;
-        }
-
-        public void run() {
-            solver.addEventListener(new SolverEventListener() {
-                @Override
-                public void bestSolutionChanged(BestSolutionChangedEvent bestSolutionChangedEvent) {
-                    scheduleManager.setSchedule((Schedule) bestSolutionChangedEvent.getNewBestSolution()); // TODO throws eaten Exception
-                }
-            });
-            solver.solve(schedule);
-            Schedule bestSchedule = (Schedule) solver.getBestSolution(); // TODO throws eaten Exception
-            scheduleManager.setSchedule(bestSchedule);
-            scheduleManager.setSolver(null);
-        }
-    }
-
+	private static final Logger LOG = LoggerFactory
+			.getLogger(ScheduleService.class);
+
+	@Inject
+	private ScheduleManager scheduleManager;
+
+	@Inject
+	private DevoxxImporter devoxxImporter;
+
+	@Inject
+	private SolverFactory solverFactory;
+
+	@Resource(name = "DefaultManagedExecutorService")
+	private ManagedExecutorService executor;
+
+	@POST
+	@Path("/import/devoxx")
+	public Response importDevoxx(@PathParam("conferenceId") Long conferenceId) {
+		Schedule schedule = devoxxImporter.importSchedule();
+		scheduleManager.setSchedule(schedule);
+		
+		StringBuffer message = new StringBuffer()
```

Comment:
This can probably be a Stringbuilder, not a StringBuffer, as it just used in a single thread?